src/shutdown/detach-dm.c

```diff
@@ -103,19 +103,27 @@ static int delete_dm(DeviceMapper *m) {
         if (fd < 0)
                 return -errno;
 
-        r = fsync_path_at(AT_FDCWD, m->path);
-        if (r < 0)
-                log_debug_errno(r, "Failed to sync DM block device %s, ignoring: %m", m->path);
-
-        return RET_NERRNO(ioctl(fd, DM_DEV_REMOVE, &(struct dm_ioctl) {
+        struct dm_ioctl dminfo = {
                 .version = {
                         DM_VERSION_MAJOR,
                         DM_VERSION_MINOR,
                         DM_VERSION_PATCHLEVEL
                 },
                 .data_size = sizeof(struct dm_ioctl),
                 .dev = m->devnum,
-        }));
+        };
+
+        if (ioctl(fd, DM_DEV_STATUS, &dminfo) < 0)
+                log_debug_errno(errno, "Failed to get DM block device %s status, ignoring: %m", m->path);
```

Comment:
hmm, this ioctl() cannot realistically fail, can it?

if so, I think we should add an "else" branch here, covering the whole if/else expression below. otherwise the log messages generated by that if/else expression will just be confusing.

Reply:
done